src/include/rear/ui.rb

```diff
@@ -184,17 +201,90 @@ def DirectoriesDialog(directories)
       deep_copy(directories)
     end
 
-    def SaveConfig(modules_load, backup_prog_include)
-      modules_load = deep_copy(modules_load)
-      backup_prog_include = deep_copy(backup_prog_include)
-      Rear.modified = true
-      Rear.output = Convert.to_string(UI.QueryWidget(Id(:output), :Value))
-      Rear.netfs_url = Convert.to_string(UI.QueryWidget(Id(:netfs_url), :Value))
-      Rear.netfs_keep_old_backup = Convert.to_boolean(
-        UI.QueryWidget(Id(:netfs_keep_old_backup), :Value)
+    def ListEditDialog(title, list)
+      # store original list for the case that the users clicks cancel
+      list_sav = list
+
+      UI.OpenDialog(
+        MinSize(
+          45,
+          15,
+          HBox(
+            HSpacing(1),
+            VBox(
+              VSpacing(0.5),
+              VBox(
+                VSquash(
+                  HBox(
+                    TextEntry(Id(:program), Opt(:notify), _("&New Entry")),
+                    VBox(
+                      VSpacing(1),
+                      PushButton(Id(:additem), Label.AddButton)
+                    )
+                  )
+                ),
+                HBox(
+                  SelectionBox(
+                    Id(:list),
+                    title,
+                    list
+                  ),
+                  Top(
+                    VBox(
+                      VSpacing(1),
+                      PushButton(Id(:delitem), Label.DeleteButton)
+                    )
+                  )
+                ),
+                ButtonBox(
+                  PushButton(Id(:ok), _("&OK")),
+                  PushButton(Id(:cancel), _("&Cancel"))
+                )
+              ),
+              VSpacing(0.5)
+            ),
+            HSpacing(1)
+          )
+        )
```

Comment:
It is an own method. Or did you mean something else?
`def ListEditDialog(title, list)`